core/lb_test.go

```diff
@@ -69,6 +69,43 @@ func TestLB_Ratchet(t *testing.T) {
 	})
 }
 
+func TestLB_SessionCleanupRace(t *testing.T) {
+	// Reproduce race condition around session cleanup #1750
+
+	assert := assert.New(t)
+	lb := NewLoadBalancingTranscoder("0", newStubTranscoder).(*LoadBalancingTranscoder)
+	sess := "sess"
+	// Force create a new session
+	_, err := lb.Transcode(stubMetadata(sess, ffmpeg.P144p30fps16x9))
+	assert.Nil(err)
+	// Mark transcoder to error out
+	transcoder := lb.sessions[sess].transcoder.(*StubTranscoder)
+	transcoder.FailTranscode = true
+
+	// Send 2 jobs concurrently to trigger the stuck issue
+	wg := newWg(2)
+	errSignal := make(chan bool)
+	// Error out on Job 1
+	go func() {
+		lb.mu.Lock() // lock the LB to prevent session cleanup
+		_, err = lb.sessions[sess].Transcode(stubMetadata(sess, ffmpeg.P144p30fps16x9))
+		assert.Equal(ErrTranscode, err)
+		errSignal <- true
+		wg.Done()
+	}()
+	// Job 2 arrives when the session loop() has closed, but session isn't cleaned up yet
+	go func() {
+		sig := <-errSignal
+		if sig {
+			_, err = lb.sessions[sess].Transcode(stubMetadata(sess, ffmpeg.P144p30fps16x9))
+			assert.Equal(ErrTranscoderStopped, err)
+		}
+		wg.Done()
+	}()
+	wg.Wait()
```

Comment:
At the moment, this test will hang upon failure. While this failure behavior will still tell us if the test breaks in the future it will also cause the entire CI process to grind to a halt. We could avoid that by updating the test to return upon a timeout. A few other tests use the [wgWait](https://github.com/livepeer/go-livepeer/blob/fd07a5673175aa6d10f56dde6eb3ef2a9f5691b9/core/orch_test.go#L204) test helper for these types of assertions. For example:

```golang
// Returns true if all WaitGroup tasks completed
// Returns false if not all WaitGroup tasks completed within the 1 second timeout
assert.True(wgWait(wg))
```